#!/usr/bin/env python3
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "sentry_chassis_controller"

gen = ParameterGenerator()

# 转向关节PID参数
gen.add("pivot_p_gain", double_t, 0, "转向关节比例增益", 1.0, 0.0, 10.0)
gen.add("pivot_i_gain", double_t, 0, "转向关节积分增益", 0.0, 0.0, 5.0)
gen.add("pivot_d_gain", double_t, 0, "转向关节微分增益", 0.0, 0.0, 2.0)
gen.add("pivot_i_max", double_t, 0, "转向关节积分上限", 0.0, 0.0, 10.0)
gen.add("pivot_i_min", double_t, 0, "转向关节积分下限", 0.0, -10.0, 0.0)

# 驱动轮PID参数
gen.add("wheel_p_gain", double_t, 0, "驱动轮比例增益", 2.0, 0.0, 20.0)
gen.add("wheel_i_gain", double_t, 0, "驱动轮积分增益", 0.1, 0.0, 5.0)
gen.add("wheel_d_gain", double_t, 0, "驱动轮微分增益", 0.0, 0.0, 2.0)
gen.add("wheel_i_max", double_t, 0, "驱动轮积分上限", 0.0, 0.0, 10.0)
gen.add("wheel_i_min", double_t, 0, "驱动轮积分下限", 0.0, -10.0, 0.0)

# 运动限制参数
gen.add("max_speed", double_t, 0, "最大线速度(m/s)", 1.0, 0.1, 9999)
gen.add("max_angular", double_t, 0, "最大角速度(rad/s)", 1.0, 0.1, 10.0)
gen.add("max_direction", double_t, 0, "最大转向角度(度)", 90.0, 10.0, 360.0)
gen.add("stop_time", double_t, 0, "停止时间阈值(s)", 0.5, 0.1, 5.0)

# 功率管理参数
gen.add("max_power", double_t, 0, "最大功率(W)", 360.0, 100.0, 1000.0)
gen.add("enable_power_limit", bool_t, 0, "启用功率限制", True)

# 调试选项
gen.add("print_expected_speed", bool_t, 0, "打印期望速度", False)
gen.add("print_expected_pivot", bool_t, 0, "打印期望转向角", False)
gen.add("print_odom", bool_t, 0, "打印里程计", False)

exit(gen.generate(PACKAGE, "sentry_chassis_controller", "SentryChassis"))
